runtime-testsuite/test/org/antlr/v4/test/runtime/go/BaseTest.java

```diff
@@ -129,36 +132,96 @@ protected void succeeded(Description description) {
 	@org.junit.Rule
 	public final Timeout eachTimeout = new Timeout(60000);
 
+	/**
+	 * Copies all files from go runtime to a temporary folder that is inside a valid GOPATH project structure.
+	 */
+	@BeforeClass
+	public static void setupGoRuntime() throws Exception {
+		tmpGopath = new File(System.getProperty("java.io.tmpdir"), "antlr-goruntime-tmpgopath-"
+				+ Long.toHexString(System.currentTimeMillis()));
+
+		ArrayList<String> pathsegments = new ArrayList<String>();
+		pathsegments.add("src");
+		pathsegments.addAll(Arrays.asList(GO_RUNTIME_IMPORT_PATH.split("/")));
+
+		File tmpPackageDir = tmpGopath;
+		for (String pathsegment : pathsegments) {
+			tmpPackageDir = new File(tmpPackageDir, pathsegment);
+		}
+		if (!tmpPackageDir.mkdirs()) {
+			throw new Exception("Could not create temp go runtime package dirs!");
+		}
+
+		File[] runtimeFiles = locateRuntime().listFiles();
+		if (runtimeFiles == null) {
+			throw new Exception("Go runtime file list is empty.");
+		}
+		for (File runtimeFile : runtimeFiles) {
+			File dest = new File(tmpPackageDir, runtimeFile.getName());
+			copyFile(runtimeFile, dest);
+		}
+
+		cacheGoRuntime(tmpPackageDir);
+	}
+
+	@AfterClass
+	public static void teardownGoRuntime() {
+		eraseDirectory(tmpGopath);
+	}
+
+	private static void cacheGoRuntime(File tmpPackageDir) throws Exception {
+		ProcessBuilder pb = new ProcessBuilder("go", "install", "-x");
```

Comment:
What does `go install -x` do? I got this on my machine:

    $ cd $GOPATH
    $ go install -x
    can't load package: package .: no buildable Go source files in /Users/Will/Developer/go